python/ray/autoscaler/_private/gcp/tpu_command_runner.py

```diff
@@ -108,15 +105,30 @@ def create_command_runner(
 
     @property
     def num_connections(self) -> int:
-        return min(self._num_workers, _MAX_NUM_CONCURRENT_ACTIVE_CONNECTIONS)
+        """Return the number of active connections allowed at a time.
+
+        We occasionally see issues where too many concurrent connections may lead to
+        failed SSH connections when there are too many TPU hosts.
+
+        We utilize this property to cap the maximum number of active connections
+        at a time until a proper fix is found.
+
+        """
+        num_max_concurrent_active_connections = os.getenv(
+            ray_constants.RAY_TPU_MAX_CONCURRENT_CONNECTIONS_ENV_VAR, default=16
```

Comment:
Ah sorry I think this might return a string and cause an error, can you double check? We seem to use `env_integer` for this https://github.com/architkulkarni/ray/blob/37c415ff380631c477afe93c1b88cd959eb00fca/python/ray/_private/ray_constants.py#L10

Reply:
Aha, you're right. Btw would you prefer that I move this altogether into ray_constants, e.g.

```
# f:ray_constants.py

RAY_TPU_MAX_CONCURRENT_CONNECTIONS = env_integer("RAY_TPU_MAX_CONCURRENT_ACTIVE_CONNECTIONS", 16)
```

(and the appropriate changes in `tpu_command_runner.py`)

Reply:
I don't have a preference about that

Reply:
Actually, let's read the env var as late as possible to make it more configurable. So the `env_integer` would be inside the command runner

Reply:
Done